packages/profiles/src/abstract/AbstractProfileManager.ts

```diff
@@ -397,20 +397,20 @@ export abstract class AbstractProfileManager<T extends IProfileTypeConfiguration
             `A request was made to save a profile of type "${this.profileType}", but no profile was supplied.`);
 
         // Ensure that the type is filled in - a mismatch will be thrown if the type indicates a type other than the manager's current type
-        parms.profile.type = parms.profile.type || this.profileType;
+        parms.type = parms.type || this.profileType;
 
         // Log the invocation
-        this.log.info(`Saving profile "${parms.profile.name}" of type "${parms.profile.type}"...`);
+        this.log.info(`Saving profile "${parms.name}" of type "${this.profileType}"...`);
```

Comment:
Should there be a ImperativeExpect check somewhere around here for if name was specified?